apps/aecore/src/aec_db.erl

```diff
@@ -411,10 +411,11 @@ transactions_by_account(AcctPubKey) ->
     ?t([T || #aec_tx{tx = T}
                  <- mnesia:index_read(aec_tx, AcctPubKey, {acct2tx})]).
 
--dialyzer({nowarn_function, transactions_by_account/2}). %% For mnesia patches.
-transactions_by_account(AcctPubKey, Filter) ->
-    ?t([T || #aec_tx{tx = T}
-        <- mnesia:index_read(aec_tx, AcctPubKey, {acct2tx}), Filter(T)]).
+-dialyzer({nowarn_function, transactions_by_account/3}). %% For mnesia patches.
+transactions_by_account(AcctPubKey, Filter, ShowPending) ->
+    ?t([T || #aec_tx{tx = T, key={_, Where}}
+        <- mnesia:index_read(aec_tx, AcctPubKey, {acct2tx}), Filter(T) 
+              andalso not (ShowPending =:= false andalso Where =:= [])]).
```

Comment:
A bit difficult to read - a helper function with descriptive name would have helped (write_tx when setting `[]` has `mempool` in the function clause that helps understanding meaning). Though looks correct.